protocols/cosi/cosi.go

```diff
@@ -289,39 +266,22 @@ func (pc *ProtocolCosi) sendChallenge(out *CosiChallenge) error {
 
 }
 
-func (pc *ProtocolCosi) StartResponse() error {
-	// TODO check the hook
-	// else do it yourself
-	resp, err := pc.Cosi.CreateResponse()
-	if err != nil {
-		return err
-	}
-	out := &CosiResponse{
-		Response: resp,
-	}
-	dbg.Lvl3(pc.Node.Name(), "ProtocolCosi().StartResponse()")
-	err = pc.SendTo(pc.Parent(), out)
-	pc.Cleanup()
-	return err
-}
-
 // handleResponse brings up the response of each node in the tree to the root.
 func (pc *ProtocolCosi) handleResponse(in *CosiResponse) error {
-	// add to temporary
-	pc.tempResponseLock.Lock()
-	pc.tempResponse = append(pc.tempResponse, in)
-	pc.tempResponseLock.Unlock()
-	// do we have enough ?
-	dbg.Lvl3(pc.Node.Name(), "has", len(pc.tempResponse), "responses")
-	if len(pc.tempResponse) < len(pc.Children()) {
-		return nil
+	if !pc.IsLeaf() {
+		// add to temporary
+		pc.tempResponseLock.Lock()
+		pc.tempResponse = append(pc.tempResponse, in)
```

Comment:
Before, lib/cosi  +protocols/cosi had specifics calls when you start a phase => `StartCommitment` or `StartReponse`. Linus just made them obsolete by passing *nil*, but it's the same thing. 
Whether one approach is better than the other one is subject to another discussion but since it looks urgent, it's OK ;)